src/util/generate-token.js

```diff
@@ -34,4 +40,6 @@ export async function generateTokenIfNot({
         browser.storage.local.set({ [USER_ID]: token.id })
         return token.id
     }
+
+    return null
```

Comment:
What happens to the calling code of this function when it returns `null`? I seem to remember from the new analytics review that it will just keep attempting to fetch a new ID (the next time events need to be sent to the server) if there is no ID set.

Reply:
It every time the server responds to failure, then it can happen. To prevent this we can just call this function some CONSTANT times. That make sense?

Reply:
There's no server request if DND is enabled. What I mean is: if you disable DND, this function will always return `null` but I remembered the calling code will just try again. 

I just found it and yes it seems like it should get into an infinite loop when it tries to send events to the server: https://github.com/mukeshkharita/WebMemex/blob/762711aedf6779cf063ca4631be8570c123a32a6/src/analytics/internal/send-to-server/send-to-server.js#L97

That part in `send-to-server` module will need to be changed in some way. Maybe in the `userId` method _(sidenote: I would change that method name so it starts with a verb. Maybe `fetchUserId`?)_, instead of looping until `userId` is defined, simply try once and throw an `Error` if it is not assigned. Then handle that in your callers (`_sendReq` and `_sendBulkReq`).

Reply:
For now, we can have just like this, but in #440 I've written the function, then we can use at both places. :slightly_smiling_face: